### ISSUE TITLE
Undefined constant NET_SSH2_COMPRESSION_ZLIB_AT_OPENSSH

### ISSUE THREAD

**Issue by michaelKaefer** [phpseclib/phpseclib#1721]
### Setup
- Docker with nginx and PHP-FPM
- phpseclib/phpseclib: 2.0.35 as well as 3.0.12 (I tried both)
- PHP 8.0.12

### Error
`Uncaught Error: Undefined constant "phpseclib\Net\NET_SSH2_COMPRESSION_ZLIB_AT_OPENSSH" at /app/vendor/phpseclib/phpseclib/phpseclib/Net/SSH2.php:4180`

Stacktrace:
```
...
#5 /app/vendor/league/flysystem-sftp/SftpConnectionProvider.php(131): League\\Flysystem\\PhpseclibV2\\SftpConnectionProvider->authenticate()
2021-12-12 21:43:04	
#4 /app/vendor/league/flysystem-sftp/SftpConnectionProvider.php(167): phpseclib\\Net\\SSH2->login()
2021-12-12 21:43:04	
#3 /app/vendor/phpseclib/phpseclib/phpseclib/Net/SSH2.php(2228): phpseclib\\Net\\SSH2->_login()
2021-12-12 21:43:04	
#2 /app/vendor/phpseclib/phpseclib/phpseclib/Net/SSH2.php(2249): phpseclib\\Net\\SSH2->_connect()
2021-12-12 21:43:04	
#1 /app/vendor/phpseclib/phpseclib/phpseclib/Net/SSH2.php(1378): phpseclib\\Net\\SSH2->_key_exchange()
2021-12-12 21:43:04	
#0 /app/vendor/phpseclib/phpseclib/phpseclib/Net/SSH2.php(1517): phpseclib\\Net\\SSH2->_send_binary_packet()
2021-12-12 21:43:04
```

### Notes
2 things are really strange:
- I get this on my production server but not on my localhost with the almost exact same setup (Docker, etc.)
- The error does not make any sense to me because the constant is defined [here](https://github.com/phpseclib/phpseclib/blob/4e16cf3f5f927a7d3f5317820af795c0366c0420/phpseclib/Net/SSH2.php#L78). I thought maybe the error is due to PHP constant namespaces (see for example this [Stackoverflow question](https://stackoverflow.com/questions/18247726/php-define-constants-inside-namespace-clarification/36007757)) but that does not seem to be the case as I cannot reproduce this.

### Conclusion
Would you accept a PR that removes the 3 `define(...)` statements in `SSH2.php` and introduces class constants instead? Maybe this would solve it, I can try this week and if it works I can create a PR. Or am I missing anything? Thank you for any help on this.

**Comment by terrafrost** [comment]
You're right - it's weird that you're getting different behavior locally vs on prod. Like even Travis CI isn't complaining about that...  weird!

But yah - class constants would work for the 2.0 / 3.0 / master branches.  The 1.0 branch isn't namespace'd so it's not an issue there. And for the 2.0 / 3.0 / master branches all the other "explicitly" defined constants are done as class constants anyway.

There is still this bit in SSH2.php (2.0 branch):

https://github.com/phpseclib/phpseclib/blob/e9fbf9ed8a6dc913c22d3fa669f44edfa7f953f8/phpseclib/Net/SSH2.php#L4450-L4462

So it's also weird that it doesn't have a problem with that whereas it does with other instances of `define()`.

**Comment by terrafrost** [comment]
TLDR feel free to do your class constants change!

If you could do a PR for the 2.0 and 3.0 branches that'd be cool! If you only want to do one then maybe target 2.0 and I can merge that into 3.0 / master and resolve any merge conflicts should there be any!

**Comment by michaelKaefer** [comment]
Thank you for your answer!

I did 3 seperate PRs for all branches:
- 2.0
- 3.0
- master

I hope that is right. Thank you!

**Comment by Roensby** [comment]
Just ran into this issue. With #1724 merged to master since last year, hopefully the fix is due to release soon.
